eva-server/src/main/java/uk/ac/ebi/eva/server/ws/RegionWSServerV2.java

```diff
@@ -0,0 +1,170 @@
+import uk.ac.ebi.eva.lib.eva_utils.MultiMongoDbFactory;
+import uk.ac.ebi.eva.server.RateLimit;
+import uk.ac.ebi.eva.server.Utils;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import static org.springframework.hateoas.mvc.ControllerLinkBuilder.linkTo;
+import static org.springframework.hateoas.mvc.ControllerLinkBuilder.methodOn;
+
+@RestController
+@RequestMapping(value = "/v2/segments")
+@Api(tags = {"segments"})
+public class RegionWSServerV2 {
+
+    private static final int REGION_REQUEST_RATE_LIMIT = 5;
+
+    @Autowired
+    private VariantWithSamplesAndAnnotationsService service;
+
+    public RegionWSServerV2() {
+    }
+
+    @GetMapping(value = "/{regionId}", produces = "application/json")
+    @ResponseBody
+    @RateLimit(value = REGION_REQUEST_RATE_LIMIT)
+    public ResponseEntity getVariantsByRegion(@PathVariable("regionId") String regionId,
+                                              @RequestParam(name = "species") String species,
+                                              @RequestParam(name = "assembly") String assembly,
+                                              @RequestParam(name = "studies", required = false) List<String> studies,
+                                              @RequestParam(name = "annot-ct", required = false) List<String>
+                                                      consequenceType,
+                                              @RequestParam(name = "maf", required = false) String maf,
+                                              @RequestParam(name = "polyphen", required = false) String polyphenScore,
+                                              @RequestParam(name = "sift", required = false) String siftScore,
+                                              @RequestParam(name = "annot-vep-version", required = false) String
+                                                      annotationVepVersion,
+                                              @RequestParam(name = "annot-vep-cache-version", required = false) String
+                                                      annotationVepCacheVersion,
+                                              HttpServletResponse response,
+                                              @ApiIgnore HttpServletRequest request)
+            throws IllegalArgumentException {
+        checkParameters(annotationVepVersion, annotationVepCacheVersion, species);
+
+        MultiMongoDbFactory.setDatabaseNameForCurrentThread(DBAdaptorConnector.getDBName(species + "_" + assembly));
+
+        List<VariantRepositoryFilter> filters = new FilterBuilder()
+                .getVariantEntityRepositoryFilters(maf, polyphenScore, siftScore, studies, consequenceType);
+
+        List<Region> regions = Region.parseRegions(regionId);
+        List<String> excludeMapped = getExcludeMapped();
```

Comment:
I need to check if this call is applicable as is now that we are changing the structure of the responses.